packages/apputils/src/sessioncontext.tsx

```diff
@@ -590,13 +643,31 @@ export class SessionContext implements ISessionContext {
         return Promise.reject(err);
       }
     }
-    const needsSelection = await this._startIfNecessary();
-    if (!needsSelection) {
-      this._isReady = true;
-      this._ready.resolve(undefined);
-    }
-    this._initPromise.resolve(needsSelection);
-    return needsSelection;
+
+    return await this._startIfNecessary();
+  }
+
+  /**
+   * Shut down the current session.
+   */
+  private async _shutdownSession(): Promise<void> {
+    const session = this._session;
+    this._session = null;
+    const kernel = session?.kernel || null;
+    this._kernelChanged.emit({
+      name: 'kernel',
+      oldValue: kernel,
+      newValue: null
+    });
+    this._statusChanged.emit('dead');
+    this._statusChanged.emit('unknown');
```

Comment:
Sanity check: is this the behavior you're looking for? To immediately (synchronously) emit the string `dead` and the string `unknown` before waiting for the shutdown promise? (Answer may be yes, I am just checking.)